Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -4001,21 +4012,32 @@ namespace AssetProcessor
         // it bundles the results up in a JobToProcessEntry struct, while it is doing this:
         JobToProcessEntry entry;
 
-        AZ::Uuid sourceUUID = AssetUtilities::GetSourceUuid(sourceAsset);
+        auto sourceUUIDOutcome = AssetUtilities::GetSourceUuid(sourceAsset);
 
-        if (sourceUUID.IsNull())
+        if (!sourceUUIDOutcome)
         {
             auto failureMessage = AZStd::string::format(
-                "CreateJobs failed for %s - Invalid UUID, metadata file " AZ_STRING_FORMAT " is likely corrupt",
-                sourceAsset.AbsolutePath().c_str(),
-                AZ_STRING_ARG(AzToolsFramework::MetadataManager::ToMetadataPath(sourceAsset.AbsolutePath().c_str()).Native()));
+                "CreateJobs failed - %s", sourceUUIDOutcome.GetError().c_str());
+
+            AZ::Uuid builderUuid = builderInfoList.size() > 0 ? builderInfoList[0].m_busId : AZ::Uuid();
```

Comment:
suggestion, for checks to determine if a list is empty or not, the `empty()` function is often more obvious.  I see other places in this file where `size()` is used so totally fine with using `size()` here.  
```suggestion
            AZ::Uuid builderUuid = builderInfoList.empty() ? AZ::Uuid() : builderInfoList[0].m_busId;
```